btrdbextras/eventproc/eventproc.py

```diff
@@ -0,0 +1,213 @@
+
+    def Register(self, name, hook, func, apikey, notify_on_success, notify_on_failure, dependencies, flags):
+        # convert decorated function to bytes
+        buff = io.BytesIO()
+        dill.dump(func, buff)
+        buff.seek(0)
+
+        params = api_pb2.RegisterRequest(
+            registration=api_pb2.Registration(
+            name=name,
+            hook=hook,
+            blob=buff.read(),
+            api_key=apikey,
+            notify_on_success=notify_on_success,
+            notify_on_failure=notify_on_failure,
+            dependencies=dependencies,
+            flags=flags,
+            )
+        )
+        response = self.stub.Register(params)
+        if hasattr(response, "handler"):
+            return Handler.from_grpc(response.handler)
+
+    def Deregister(self, handler_id):
+        params = api_pb2.DeregisterRequest(id=handler_id)
+        response = self.stub.Deregister(params)
+        return response
+
+
+##########################################################################
+## Public Functions
+##########################################################################
+
+def hooks(conn):
+    """
+    List registered hooks.
+
+    Parameters
+    ----------
+    conn: Connection
+        btrdbextras Connection object containing a valid address and api key.
+    """
+    s = Service(connect(conn))
+    return s.ListHooks()
+
+def list_handlers(conn, hook=""):
+    """
+    List registered handlers.  An optional hook name is allowed to filter
+    results.
+
+    Parameters
+    ----------
+    conn: Connection
+        btrdbextras Connection object containing a valid address and api key.
+    hook: str
+        Optional hook name to filter registered handlers.
+
+    """
+    s = Service(connect(conn))
+    return [Handler.from_grpc(h) for h in s.ListHandlers(hook)]
+
+
+def deregister(conn, handler_id):
+    """
+    Removes an existing event handler by ID.
+
+    Parameters
+    ----------
+    conn: Connection
+        btrdbextras Connection object containing a valid address and api key.
+    handler_id: int
+        ID of the event handler to remove.
+
+    """
+    s = Service(connect(conn))
+    h = s.Deregister(handler_id)
+    return h.id == handler_id
+
+
+def register(conn, name, hook, notify_on_success, notify_on_failure, dependencies="", flags=None):
+    """
+    decorator to submit (register) an event handler function
+
+    Parameters
+    ----------
+    conn: Connection
+        btrdbextras Connection object containing a valid address and api key.
+    name: str
+        Friendly name of this event handler for display purposes.
+    hook: str
+        Name of the hook that this event handler responds to.
+    notify_on_success: str
+        Email address of user to notify when event handler completes successfully.
+    notify_on_failure: str
+        Email address of user to notify when event handler does not complete
+        successfully.
+    dependencies: str
```

Comment:
I'd remove this for now, since we don't have support for doing anything with this yet. It's a kwarg, so we can add it back later without a breaking cahnge.

Reply:
I've removed from func signature and docs but left in a placeholder.